File: JavaPools/PoolsLibrary/src/main/java/ru/urfu/javapools/poolslibrary/pool/basicfunctionality/item/ObjectCreationFailedException.java
package ru.urfu.javapools.poolslibrary.pool.basicfunctionality.item;

import ru.urfu.javapools.poolslibrary.controller.PoolException;
import ru.urfu.javapools.poolslibrary.function.FunctionThatMayThrow;

@SuppressWarnings({"serial", "rawtypes"})
public class ObjectCreationFailedException extends PoolException {
	
	private FunctionThatMayThrow _usedCreateDelegate;
	
	private static final String _MESSAGE_PATTERN = "Creation of object with key='%s' failed. Look at cause for details";
	
	public ObjectCreationFailedException (Object key, FunctionThatMayThrow usedCreateDelegate, Throwable cause) {
				
		super(key, String.format(_MESSAGE_PATTERN, key), cause);		
		_usedCreateDelegate = usedCreateDelegate;		
	}
	
	public FunctionThatMayThrow getUsedCreateDelegate() {
		return _usedCreateDelegate;
	}
}

File: CSharpPools/PoolsLibrary/Pool/BasicFunctionality/Item/ObjectCreationFailedException.cs
using System;
using PoolsLibrary.Controller;

namespace PoolsLibrary.Pool.BasicFunctionality.Item
{
    public class ObjectCreationFailedException<TK, TV> : PoolException<TK>
    {
        public Func<TK, TV> UsedCreateDelegate { get; private set; }

        public ObjectCreationFailedException(TK key, Func<TK,TV> usedCreateDelegate, string message = null, Exception innerException = null)
            : base(key, message, innerException)
        {
            UsedCreateDelegate = usedCreateDelegate;
        }

        public ObjectCreationFailedException()
        {
        }
    }
}
